src/REPLMode/REPLMode.jl

```diff
@@ -692,15 +693,72 @@ function try_prompt_pkg_add(pkgs::Vector{Symbol})
         println(ctx.io)
         return false
     end
-    if lowercase(resp) in ["y", "yes"]
-        API.add(string.(available_pkgs))
+    function handle_resp(resp, called_from_help)
```

Comment:
Having this as a local function (closure?) invites complexity.
Makes a long function even longer
 Better to move it out of this, even if it does mean passing in a few more things.